lib/queuePopulator/QueuePopulator.js

```diff
@@ -51,18 +56,29 @@ class QueuePopulator {
      */
     open(cb) {
         this._loadExtensions();
-        this._setupZookeeper(err => {
+        async.series([
+            next => this._setupZookeeper(next),
+            next => process.nextTick(() => {
```

Comment:
You can avoid an unnecessary event loop iteration by integrating the `setupLogSources()` call into the next function in the series, just before calling `this._setupMetricsProducer()`, it will also slightly enhance clarity.

Reply:
Do you mean to add it to the series with `next => this._setupLogSources(next)`?
If so, I'll be adding a callback to `setupLogSources` which just gets invoked at the end of its function?